ui/src/components/WMWorkerPoolEditor/index.jsx

```diff
@@ -0,0 +1,321 @@
+import SpeedDialAction from '../SpeedDialAction';
+import SpeedDial from '../SpeedDial';
+
+@withRouter
+@withStyles(theme => ({
+  saveIcon: {
+    ...theme.mixins.successIcon,
+  },
+  deleteIcon: {
+    ...theme.mixins.errorIcon,
+  },
+  createIconSpan: {
+    ...theme.mixins.fab,
+    ...theme.mixins.actionButton,
+  },
+  saveIconSpan: {
+    position: 'fixed',
+    bottom: theme.spacing.double,
+    right: theme.spacing.unit * 11,
+    ...theme.mixins.actionButton,
+  },
+  dropdown: {
+    minWidth: 200,
+  },
+  list: {
+    paddingLeft: 0,
+    paddingRight: 0,
+  },
+  middleList: {
+    paddingTop: theme.spacing.unit * 7,
+    paddingBottom: theme.spacing.unit * 9,
+    paddingLeft: 0,
+    paddingRight: 0,
+  },
+  separator: {
+    padding: theme.spacing.double,
+    paddingBottom: 0,
+  },
+}))
+export default class WMWorkerPoolEditor extends Component {
+  static defaultProps = {
+    allowEditWorkerPoolId: false,
+    providerType: GCP,
+  };
+
+  static propTypes = {
+    workerPool: WorkerManagerWorkerPoolSummary.isRequired,
+    saveRequest: func.isRequired,
+    providerType: string.isRequired,
+    allowEditWorkerPoolId: bool,
+    deleteRequest: func,
+  };
+
+  state = {
+    workerPool: {
+      workerPoolId1: this.props.workerPool.workerPoolId1,
+      workerPoolId2: this.props.workerPool.workerPoolId2,
+      description: this.props.workerPool.description,
+      owner: this.props.workerPool.owner,
+      emailOnError: this.props.workerPool.emailOnError,
+      config: this.props.workerPool.config,
+    },
+    providerType: this.props.providerType,
+    invalidProviderConfig: false,
+    actionLoading: false,
+    error: null,
+  };
+
+  handleInputChange = ({ target: { name, value } }) => {
+    this.setState({ workerPool: { ...this.state.workerPool, [name]: value } });
+  };
+
+  handleSwitchChange = event => {
+    const {
+      target: { value },
+    } = event;
+
+    this.setState({
+      workerPool: {
+        ...this.state.workerPool,
+        [value]: !this.state.workerPool[value],
+      },
+    });
+  };
+
+  handleProviderTypeChange = event => {
+    const {
+      target: { value },
+    } = event;
+
+    this.setState({
+      providerType: value,
+      workerPool: {
+        ...this.state.workerPool,
+        config: PROVIDER_CONFIGS.get(value),
+      },
+    });
+  };
+
+  handleEditorChange = value => {
+    const { workerPool } = this.state;
+
+    try {
+      workerPool.config = JSON.parse(value);
+
+      this.setState({
+        workerPool,
+        invalidProviderConfig: false,
+      });
+    } catch (err) {
+      workerPool.config = value;
+
+      this.setState({
+        workerPool,
+        invalidProviderConfig: true,
+      });
+    }
+  };
+
+  handleOnClick = async event => {
+    const { workerPoolId1, workerPoolId2, ...payload } = this.state.workerPool;
+    const { name: requestName } = event.currentTarget;
+
+    payload.providerId = PROVIDERS.get(this.state.providerType);
```

Comment:
> I guess this information will eventually come from the API?

Maybe. I would try to minimize the network calls, but we'll see how the API will work. I would like it when opening the editor we could get a list of existing providers, so that we know the ID and type of each. `Provider Type` in the UI would be replaced with `Provider`, and I'd still have the `PROVIDER_CONFIGS` constant